lib/scorecard.rb

```diff
@@ -0,0 +1,53 @@
+require 'frame.rb'
+
+class Scorecard
+
+  attr_accessor :frames, :total_score
+
+  def initialize(frames = Frame.new )
+    @frames = [frames]
+    @current_turn = 0
+    @total_score = 0
+  end
+
+  def roll(pins)
+    if @frames[turn].next_turn?
+      create_new_frame
+      @frames[turn].rolled(pins)
+      calculate_total
+    else
+      @frames[turn].rolled(pins)
+      calculate_total
+    end
+  end
+
+  def turn
+    @current_turn
+  end
+
+  def calculate_total
+    @total_score = 0
```

Comment:
I suspect there may be issues setting to zero here and in the initialize method - I guess whether it should be set to zero here depends on whether this method is a running total or the total just by each frame